modules/restore-add-master.adoc

```diff
@@ -0,0 +1,77 @@
+// Module included in the following assemblies:
+//
+// * backup_and_restore/replacing-failed-master.adoc
+
+[id="restore-add-master_{context}"]
+= Adding a master host back to the etcd cluster
+
+Follow these steps to add a master host back to the etcd cluster. This procedure assumes that you previously removed the master host from the cluster and that its TLS certificates are valid.
+
+// TODO: Ask why it can't be a completely new member?
+
+.Prerequisites
+
+* Access to the cluster as a user with the `cluster-admin` role.
+* SSH access to the master host to add to the etcd cluster.
+* The IP address of an existing active etcd member.
+
+.Procedure
+
+. Access the master host to add to the etcd cluster.
+
+. Run the `etcd-member-add.sh` script.
++
+Pass in two parameters to the `etcd-member-add.sh` script: IP address of an existing etcd member and the name of the etcd member to add.
++
+----
+[core@ip-10-0-128-73 ~]$ sudo -E etcd-member-add.sh 10.0.128.73 \ etcd-member-ip-10-0-147-172.us-east-2.compute.internal
+Downloading etcdctl binary..
+etcdctl version: 3.3.10
+API version: 3.3
+etcd-member.yaml found in ./assets/backup/
+etcd.conf backup upready exists ./assets/backup/etcd.conf
+Stopping etcd..
+Waiting for etcd-member to stop
+etcd data-dir backup found ./assets/backup/etcd..
+Updating etcd membership..
+Removing etcd data_dir /var/lib/etcd..
+
+ETCD_NAME="etcd-member-ip-10-0-147-172.us-east-2.compute.internal"
+ETCD_INITIAL_CLUSTER="etcd-member-ip-10-0-147-172.us-east-2.compute.internal=https://etcd-1.clustername.devcluster.openshift.com:2380,etcd-member-ip-10-0-171-108.us-east-2.compute.internal=https://etcd-2.clustername.devcluster.openshift.com:2380,etcd-member-ip-10-0-128-73.us-east-2.compute.internal=https://etcd-0.clustername.devcluster.openshift.com:2380"
+ETCD_INITIAL_ADVERTISE_PEER_URLS="https://etcd-1.clustername.devcluster.openshift.com:2380"
+ETCD_INITIAL_CLUSTER_STATE="existing"'
+Member  1e42c7070decd39 added to cluster 6e25bab1bb556673
+Starting etcd..
+----
++
+// TODO: Didn't include the pod list verification - okay to skip?
```

Comment:
@retroflexer I didn't include the step to verify the list of pods, because I figured that the next step takes care of the verification. Let me know if you do think that it is a critical to add.

Reply:
`member list` is different from ` oc get pods -n openshift-etcd`. We need to verify both.

Reply:
To add some context: member list will only show if the member has actually started and joined the cluster in the past. Even if now the member is not running it would still report `started` vs `unstarted`. To verify the health of the members we can do `etcdctl endpoint health --cluster` in which it should return healthy for all 3 members. I think it is reasonable to also verify the pods are running as Suresh pointed out.